### PR TITLE
Fix autoscaler Lambda download

### LINKED ISSUE #56: Archive creation error
I'm attempting to upgrade from 1.5 to 1.8.1 and running into issues:

```
Terraform used the selected providers to generate the following execution
plan. Resource actions are indicated with the following symbols:
  ~ update in-place

Terraform planned the following actions, but then encountered a problem:

  # module.this.aws_iam_instance_profile.this will be updated in-place
  ~ resource "aws_iam_instance_profile" "this" {
        id          = "sp5ft-XXX"
        name        = "sp5ft-XXX"
        tags        = {}
      ~ tags_all    = {} -> (known after apply)
        # (5 unchanged attributes hidden)
    }

  # module.this.aws_iam_role.this[0] will be updated in-place
  ~ resource "aws_iam_role" "this" {
        id                    = "sp5ft-XXX"
        name                  = "sp5ft-XXX"
        tags                  = {}
      ~ tags_all              = {} -> (known after apply)
        # (8 unchanged attributes hidden)

        # (1 unchanged block hidden)
    }

Plan: 0 to add, 2 to change, 0 to destroy.
╷
│ Error: Archive creation error
│ 
│   with module.this.data.archive_file.binary[0],
│   on .terraform/modules/this/autoscaler.tf line 19, in data "archive_file" "binary":
│   19: data "archive_file" "binary" {
│ 
│ error creating archive: error archiving file: could not archive missing
│ file: lambda/bootstrap
╵
[01HCAE4P5SYQE9H2BX0K9P4FBM] Unexpected exit code when planning changes: 1
```

### PR DESCRIPTION
**Pull request by alexjurkiewicz**
Fixes #56.

### CODE REVIEW

**Comment by eminalemdar** [comment]
hello @alexjurkiewicz, thank you for creating this PR. can you explain why the resource type `null_resource` is not working for you?

**Comment by alexjurkiewicz** [comment]
`null_resource` should work fine if you want to preserve compatibility with older TF versions 👍 I prefer modules to minimise their provider dependencies -- `terraform init` is faster

File: autoscaler.tf [hunk 1]
```diff
@@ -9,8 +9,12 @@ resource "aws_ssm_parameter" "spacelift_api_key_secret" {
   value = var.spacelift_api_key_secret
 }
 
-resource "null_resource" "download" {
+resource "terraform_data" "download" {
   count = var.enable_autoscaling ? 1 : 0
+  triggers_replace = {
```
Comment (alexjurkiewicz) [inline autoscaler.tf:14]:
```suggestion
resource "null_resource" "download" {
  count = var.enable_autoscaling ? 1 : 0
  triggers = {
```

**Comment by alexjurkiewicz** [comment]
Converted to `null_resource` as discussed 😊

File: autoscaler.tf [hunk 2]
```diff
@@ -21,7 +25,7 @@ data "archive_file" "binary" {
   type        = "zip"
   source_file = "lambda/bootstrap"
   output_path = "ec2-workerpool-autoscaler_${var.autoscaler_version}.zip"
-  depends_on  = [null_resource.download]
+  depends_on  = [terraform_data.download]
```
Comment (jeohist) [inline autoscaler.tf:28]: Should be `null_resource.download`?

Comment (alexjurkiewicz) [reply autoscaler.tf:28]:
```suggestion
  depends_on  = [null_resource.download]
```

**Comment by jeohist** [comment]
Although the file is relatively small, perhaps Spacelift could publish the hash as an asset of the release so we can use that as a trigger (get hash with HTTP provider -> use as trigger for null_resource)? Then it's unnecessary to download the file every time, only when it's changed.

**Comment by alexjurkiewicz** [comment]
You could probably rig something up with fileexists:
https://developer.hashicorp.com/terraform/language/functions/fileexists

However, if you're using this module, you're running Terraform in
Spacelift. And in Spacelift, there's no persistent filesystem state. You'll
be downloading the module every time anyway.

File: providers.tf [hunk 3]
```diff
@@ -5,4 +5,5 @@ terraform {
       version = ">= 4.0"
     }
   }
+  required_version = ">= 1.4.0"
```
Comment (jmfontaine) [inline providers.tf:8]: This seems like a leftover of the `terraform_data` removal. I would recommend removing it too.

**Review by jmfontaine — CHANGES_REQUESTED** [review]
Other than the recommend minor change, it looks good to me.

File: providers.tf (hunk 3 above)
Comment (alexjurkiewicz) [reply providers.tf:8]:
```suggestion
```

Comment (alexjurkiewicz) [reply providers.tf]: Thanks, done

**Review by jmfontaine — APPROVED** [review]
Thanks @alexjurkiewicz!

We do not allow checks for forks for security reasons. @eminalemdar will figure out how to merge your contribution.

**Review by eminalemdar — CHANGES_REQUESTED** [review]
to be able to introduce these changes the module version should be bumped as well both in the config and the documentation section. can you please also update those?

**Comment by alexjurkiewicz** [comment]
Updated this PR

**Review by eminalemdar — CHANGES_REQUESTED** [review]
hey @alexjurkiewicz, you also need to update the version number in the example module block in the `README` file to match the `config.yml`. can you please update that as well and also documentation update will be required after that change.

**Review by eminalemdar — APPROVED** [review]

**Pull request merged**